src/rqt_graph/dotcode.py

```diff
@@ -226,19 +226,27 @@ def _add_edge(self, edge, dotcode_factory, dotgraph, is_topic=False):
                     label=temp_label,
                     penwidth=penwidth,
                     color=color,
-                    edgetooltip=self._qos_to_string(edge.qos))
+                    edgetooltip=self._qos_to_string(edge.start, edge.end, edge.qos))
             else:
                 dotcode_factory.add_edge_to_graph(
                     dotgraph,
                     _conv(edge.start),
                     _conv(edge.end),
                     label=edge.label,
-                    edgetooltip=self._qos_to_string(edge.qos))
+                    edgetooltip=self._qos_to_string(edge.start, edge.end, edge.qos))
 
-    def _qos_to_string(self, qos):
+    def _qos_to_string(self, start, end, qos):
         if qos is None:
             return None
-        s = 'QoS settings'
+        if start[0] == ' ':
```

Comment:
Topic names start with a single whitespace character? Or should this be `/`? Seems a bit brittle, but okay.

Reply:
I was also surprised, but that's an old hack in `rqt_graph` code https://github.com/ros-visualization/rqt_graph/blob/6d87edcdc6283fda115dc721dad9218e47b0a8e2/src/rqt_graph/rosgraph2_impl.py#L67 

I didn't do it :joy:

Reply:
I can probably modify the `qos` argument to be already the parsed text, so we don't need this hack

Reply:
At the very least, maybe just add a comment pointing to that line so future readers aren't confused. We can also do some refactoring in a follow-up PR if you like.

Reply:
Added comment in https://github.com/ros-visualization/rqt_graph/pull/60/commits/85c9a24d2d56982d0cbd6f0a27748743913a2976.
I will try to refactor in a follow-up.